coderd/database/queries/workspaces.sql

```diff
@@ -266,22 +271,42 @@ WHERE
 	END
 	-- Authorize Filter clause will be injected below in GetAuthorizedWorkspaces
 	-- @authorize_filter
-ORDER BY
-	-- To ensure that 'favorite' workspaces show up first in the list only for their owner.
-	CASE WHEN workspaces.owner_id = @requester_id AND workspaces.favorite THEN 0 ELSE 1 END ASC,
-	(latest_build.completed_at IS NOT NULL AND
-		latest_build.canceled_at IS NULL AND
-		latest_build.error IS NULL AND
-		latest_build.transition = 'start'::workspace_transition) DESC,
-	LOWER(users.username) ASC,
-	LOWER(workspaces.name) ASC
-LIMIT
-	CASE
-		WHEN @limit_ :: integer > 0 THEN
-			@limit_
-	END
-OFFSET
-	@offset_
+), filtered_workspaces_order AS (
+	SELECT
+		fw.*
+	FROM
+		filtered_workspaces fw
+	ORDER BY
+		-- To ensure that 'favorite' workspaces show up first in the list only for their owner.
+		CASE WHEN owner_id = @requester_id AND favorite THEN 0 ELSE 1 END ASC,
+		(latest_build_completed_at IS NOT NULL AND
+			latest_build_canceled_at IS NULL AND
+			latest_build_error IS NULL AND
+			latest_build_transition = 'start'::workspace_transition) DESC,
+		LOWER(username) ASC,
+		LOWER(name) ASC
+	LIMIT
+		CASE
+			WHEN @limit_ :: integer > 0 THEN
+				@limit_
+		END
+	OFFSET
+		@offset_
+), total_count AS (
+	SELECT
+		count(*) AS count
+    FROM
+		filtered_workspaces
+)
+SELECT
+	tc.count,
+	fwo.*
+FROM
+	total_count tc
+LEFT JOIN
+	filtered_workspaces_order fwo
```

Comment:
@johnstcn @mafredri 

I've applied the solution we chatted yesterday, and it works well on the SQL level, but when it comes to `model` structures, `sqlc` converted all properties of `GetWorkspaceRows` to nullables (see CI). The dumb solution would be to coalesce all properties with "zero" values, but wouldn't it be too dirty?

Reply:
An ugly hack would be to ensure there is always one empty row of workspace and use inner join instead of outer:

```
with filtered_workspaces_order as (
  select ...
  union all
  select '', '', ...
)
```

Maybe sqlc.embed could help too?

Reply:
Whenever I've had issues with `sqlc` 'incorrectly' inferring a nullable type, I've always worked around it by adding explicit type casts.

Something like:

```
	fwo.id :: uuid as id,
	fwo.created_at :: timestamptz as created_at,
	fwo.updated_at :: timestamptz as updated_at,
	fwo.owner_id :: uuid as owner_id,
	fwo.organization_id :: uuid as organization_id,
	fwo.template_id :: uuid as template_id,
	fwo.deleted :: boolean as deleted,
	fwo.name :: text as name,
	fwo.autostart_schedule as autostart_schedule, -- nullable so no typecast
	fwo.ttl as ttl, -- nullable so no typecast
	fwo.last_used_at :: timestamptz as last_used_at,
	fwo.dormant_at as dormant_at, -- nullable so no typecast
	fwo.deleting_at as deleting_at, -- nullable so no typecast
	fwo.automatic_updates :: automatic_updates as automatic_updates,
	fwo.favorite :: boolean as favorite,
	fwo.template_name :: text as template_name,
	fwo.template_version_id :: uuid as template_version_id,
	fwo.template_version_name :: text as template_version_name,
	fwo.username :: text as username,
	fwo.latest_build_completed_at :: timestamptz as latest_build_completed_at,
	latest_build_canceled_at :: timestamptz as latest_build_canceled_at,
	latest_build_error :: text as latest_build_error,
	latest_build_transition :: workspace_transition as latest_build_transition
```

Reply:
Also note that you will need to manually update the scan order of the `GetAuthorizedWorkspaces` query when you make any changes to `GetWorkspaces`

Reply:
I might have sorted it out 👍 The query does not look great but works. I will work on tests and polishing. Thanks for guidance!